### PR TITLE
Rewrite all documentation on how to build Crosswalk

### PR DESCRIPTION
**Pull request by rakuco**
The existing page had too much content about too many different
platforms, making it confusing and hard to follow.

The new version splits everything up into different pages that cover
different topics:
* "Prerequisites" covers proxy setup and fetching depot_tools.
* The per-platform "building" pages explain how to fetch the source and
  build Crosswalk for Android, Linux and Windows, with information about
  both gyp and GN and the required dependencies for each OS. The Windows
  page also covers Visual Studio integration.
* "Running tests" covers how to run Crosswalk's tests on all platforms.
* "Tracking branches" covers tracking a Crosswalk branch other than
  master.

### CODE REVIEW

File: public/contribute/building_crosswalk/tracking_branches.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+## Set the branch before the initial checkout
+
+If you haven't checked out Crosswalk's source code yet, just pass the URL of
+the branch you want to checkout to the `gclient config` call. For example, to
+track the _crosswalk-2_ branch:
+
+    gclient config --name=src/xwalk \
+      https://github.com/crosswalk-project/crosswalk.git@origin/crosswalk-2
+
+## Change the branch for an existing checkout*
```
Comment (baleboy) [inline public/contribute/building_crosswalk/tracking_branches.md]: Is the asterisk meant to be there? I didn't see any footnote

File: public/contribute/building_crosswalk.md [hunk 2]
```diff
@@ -1,282 +1,33 @@
...
+The instructions below correspond to the operating system you are using to
+_build_ Crosswalk (the "host"), not the platform you are targeting. That is, if
+you want to build Crosswalk for Android, you still need to follow the
+instructions for Linux below.
+
+1. [Prerequisites](building_crosswalk/prerequisites.html) covers dealing with
+   network proxies and correctly installing and bootstrapping `depot_tools`.
+1. The platform-specific pages detail how to check out and build Crosswalk for
+   different operating systems:
+   * [Android](android_build.html)
```
Comment (baleboy) [inline public/contribute/building_crosswalk.md]: the link is missing the building_crosswalk path

File: public/contribute/building_crosswalk.md [hunk 3]
```diff
@@ -1,282 +1,33 @@
...
+   * [Linux](linux_build.html)
```
Comment (baleboy) [inline public/contribute/building_crosswalk.md]: ditto

File: public/contribute/building_crosswalk.md [hunk 4]
```diff
@@ -1,282 +1,33 @@
...
+   * [Windows](windows_build.html)
```
Comment (baleboy) [inline public/contribute/building_crosswalk.md]: ditto

File: public/contribute/building_crosswalk.md [hunk 5]
```diff
@@ -1,282 +1,33 @@
...
-
-Then run them with:
-
-    ./out/Release/xwalk_unitttest
-    ./out/Release/xwalk_browsertest
+The Crosswalk Project is structured around and heavily depends on the
+[Chromium project](https://www.chromium.org/Home).
+
+As such, Crosswalk uses the same build system and auxiliary tools as Chromium:
+* `depot_tools` manages source code checkouts, the dependencies among them and
```
Comment (baleboy) [inline public/contribute/building_crosswalk.md]: this didn't show up as a bullet point in my machine. Instead this paragraph was all italic

File: public/contribute/building_crosswalk.md [hunk 6]
```diff
@@ -1,282 +1,33 @@
...
+  ships some additional tools required to build Chromium and Crosswalk (such as
+  ninja on all platforms and git and Python on Windows).
+* `ninja` is the actual build system, responsible for invoking the compiler,
```
Comment (baleboy) [inline public/contribute/building_crosswalk.md]: as above

**Comment by rakuco** [comment]
Patch rebased, I think all the issues have been addressed.
